WordPress/Classes/ViewRelated/Reader/ReaderMenuViewController.swift

```diff
@@ -0,0 +1,172 @@
+
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+    }
+
+
+    // MARK: - Instance Methods
+
+
+    /// Presents the detail view controller for the specified post on the specified
+    /// blog. This is a convenience method for use with Notifications (for example).
+    ///
+    /// - Parameters:
+    ///     - postID: The ID of the post on the specified blog.
+    ///     - blogID: The ID of the blog.
+    ///
+    func openPost(postID: NSNumber, onBlog blogID: NSNumber) {
+        let controller = ReaderDetailViewController.controllerWithPostID(postID, siteID: blogID)
+        navigationController?.pushViewController(controller, animated: true)
+    }
+
+
+    /// Presents the post list for the specified topic.
+    ///
+    /// - Parameters:
+    ///     - topic: The topic to show.
+    ///
+    func showPostsForTopic(topic: ReaderAbstractTopic) {
+        let controller = ReaderStreamViewController.controllerWithTopic(topic)
+        navigationController?.pushViewController(controller, animated: true)
+    }
+
+
+    /// Presents the reader's search view controller.
+    ///
+    func showReaderSearch() {
+        let controller = ReaderSearchViewController.controller()
+        navigationController?.pushViewController(controller, animated: true)
+    }
+
+
+    // MARK: - TableView Delegate Methods
+
+
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.numberOfSectionsInMenu()
+    }
+
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfItemsInSection(section)
+    }
+
+
+    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let title = viewModel.titleForSection(section)
+        let header = WPTableViewSectionHeaderFooterView(reuseIdentifier: nil, style: .Header)
+        header.title = title
+        return header
+    }
+
+
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        let title = viewModel.titleForSection(section)
+        return WPTableViewSectionHeaderFooterView.heightForHeader(title, width: view.frame.width)
+    }
+
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(cellIdentifier)!
+        configureCell(cell, atIndexPath: indexPath)
+        return cell
+    }
+
+
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        guard let menuItem = viewModel.menuItemAtIndexPath(indexPath) else {
+            return
+        }
+
+        // TODO: Remember selection
+
+        if let topic = menuItem.topic {
+            showPostsForTopic(topic)
+            return
+        }
+
+        if menuItem.type == .Search {
```

Comment:
Thought: Maybe `type = .Topic` could have a topic associated, so that you could do...

```
switch menuItem.type {
   case .Topic(let topic):
   case .Search:
}
```

(Just a thought, not sure how that would fit in the bigger picture)